graspologic/inference/ldt.py

```diff
@@ -0,0 +1,416 @@
+        raise TypeError(msg)
+
+    if input_graph:
+        A1 = import_graph(A1)
+        A2 = import_graph(A2)
+
+        X1_hat, X2_hat = _embed(A1, A2, n_components)
+    else:
+        # check for nx objects, since they are castable to arrays,
+        # but we don't want that
+        if not isinstance(A1, np.ndarray):
+            msg = (
+                f"Embedding of the first graph is of type {type(A1)}, not "
+                "np.ndarray. If input_graph is False, the inputs need to be "
+                "adjacency spectral embeddings, with shapes (n, d) and "
+                "(m, d), passed as np.ndarrays."
+            )
+            raise TypeError(msg)
+        if not isinstance(A2, np.ndarray):
+            msg = (
+                f"Embedding of the second graph is of type {type(A2)}, not an "
+                "array. If input_graph is False, the inputs need to be "
+                "adjacency spectral embeddings, with shapes (n, d) and "
+                "(m, d), passed as np.ndarrays."
+            )
+            raise TypeError(msg)
+
+        if A1.ndim != 2:
+            msg = (
+                "Embedding array of the first graph does not have two dimensions. "
+                "If input_graph is False, the inputs need to be adjacency "
+                "spectral embeddings, with shapes (n, d) and (m, d)"
+            )
+            raise ValueError(msg)
+        if A2.ndim != 2:
+            msg = (
+                "Embedding array of the second graph does not have two dimensions. "
+                "If input_graph is False, the inputs need to be adjacency "
+                "spectral embeddings, with shapes (n, d) and (m, d)"
+            )
+            raise ValueError(msg)
+        if A1.shape[1] != A2.shape[1]:
+            msg = (
+                "Two input embeddings have different number of components. "
+                "If input_graph is False, the inputs need to be adjacency "
+                "spectral embeddings, with shapes (n, d) and (m, d)"
+            )
+            raise ValueError(msg)
+
+        # checking for inf values
+        X1_hat = check_array(A1)
+        X2_hat = check_array(A2)
+
+    if align_type == "sign_flips":
+        aligner = SignFlips(**align_kws)
+        X1_hat = aligner.fit_transform(X1_hat, X2_hat)
+    elif align_type == "seedless_procrustes":
+        aligner = SeedlessProcrustes(**align_kws)
+        X1_hat = aligner.fit_transform(X1_hat, X2_hat)
+
+    if size_correction:
+        X1_hat, X2_hat = _sample_modified_ase(X1_hat, X2_hat, pooled=pooled)
+
+    metric_func_ = _instantiate_metric_func(metric=metric, test=test)
+    test_obj = KSample(test, compute_distance=metric_func_)
+
+    data = test_obj.test(X1_hat, X2_hat, reps=n_bootstraps, workers=workers, auto=False)
+
+    null_distribution_ = test_obj.indep_test.null_dist
+    sample_T_statistic_ = data[0]
+    p_value_ = data[1]
+
+    return p_value_, sample_T_statistic_, null_distribution_
```

Comment:
@bdpedigo it feels like we planned to return something else, but i cannot think of what could that me. Q? metric?

Reply:
d?